python/cudf_polars/tests/expressions/test_booleanfunction.py

```diff
@@ -0,0 +1,129 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import pytest
+
+import polars as pl
+
+from cudf_polars.testing.asserts import assert_gpu_result_equal
+
+
+@pytest.fixture(params=[False, True], ids=["no_nulls", "nulls"])
+def has_nulls(request):
+    return request.param
+
+
+@pytest.mark.parametrize(
+    "ignore_nulls",
+    [
+        pytest.param(
+            False, marks=pytest.mark.xfail(reason="No support for Kleene logic")
+        ),
+        True,
+    ],
+)
+def test_booleanfunction_reduction(ignore_nulls):
+    ldf = pl.LazyFrame(
+        {
+            "a": [1, 2, 3.0, 2, 5],
+            "b": [0, 3, 1, -1, None],
+            "c": [1, 6, 5, 3, 2],
+        }
+    )
+
+    query = ldf.select(
+        (pl.col("a") > 3).any(ignore_nulls=ignore_nulls),
+        (pl.col("b") > 2).all(ignore_nulls=ignore_nulls),
+    )
+
+    assert_gpu_result_equal(query)
+
+
+@pytest.mark.parametrize(
+    "expr",
+    [
+        pl.Expr.is_null,
+        pl.Expr.is_not_null,
+        pl.Expr.is_nan,
+        pl.Expr.is_not_nan,
+    ],
+    ids=lambda f: f"{f.__name__}()",
+)
+@pytest.mark.parametrize("has_nans", [False, True], ids=["no_nans", "nans"])
+def test_boolean_function_unary(request, expr, has_nans, has_nulls):
+    if has_nulls and expr in (pl.Expr.is_nan, pl.Expr.is_not_nan):
+        request.applymarker(
```

Comment:
I suppose this avoids the whole programmatic xfail issue, TIL.

Reply:
it's in the dev docs! https://docs.rapids.ai/api/cudf/nightly/developer_guide/testing/#conditional-xfails